routes/users.js

```diff
@@ -994,9 +994,29 @@ router.get('/user/:id/activate', async function(req, res) {
 
     let data = await utils.mongo_groups().findOne({'name': user.group});
     if(!data) {
-        res.status(403).send('Group ' + user.group + ' does not exist, please create it first');
-        res.end();
-        return;
+        if (CONFIG.general.auto_add_group) {
+            let mingid = await utils.getGroupAvailableId();
```

Comment:
does this behavior not already done somewhere else to call a function rather than "recoding" this (and avoid later issues) ?

The behavior is optional by config, but I would highly recommend not to do that (return of experience...).
Before, we used to do so, and lead to many errors in group names. Some people were in the same functional "group" but with different unix groups (  cnrsXX, cnrs_XX, umrXX, etc...) . Admin used to click "quickly" on activate without strict checking of group names.
This is usually only discover quite some time after, and we need to delete some groups, move users etc....

Reply:
> does this behavior not already done somewhere else to call a function rather than "recoding" this (and avoid later issues) ?
> 

Yes we can factorize with https://github.com/genouest/genouestaccountmanager/blob/35d61c084ecd9735560bcc05a26b13b90c3abb77/routes/users.js#L541

> The behavior is optional by config, but I would highly recommend not to do that (return of experience...).
> Before, we used to do so, and lead to many errors in group names. Some people were in the same functional "group" but with different unix groups ( cnrsXX, cnrs_XX, umrXX, etc...) . Admin used to click "quickly" on activate without strict checking of group names.
> This is usually only discover quite some time after, and we need to delete some groups, move users etc....

Yes I know, I have another PR in preparation to solve this issue, it will allow to use other value for default group (instead of team, use username or a specific group)

Reply:
Factorization done, I am testing on my dev env right now.